csharp/src/Ice/TcpEndpoint.cs

```diff
@@ -194,18 +208,38 @@ internal TcpEndpoint(InputStream istr, Protocol protocol)
             string connectionId,
             bool compressionFlag,
             int timeout) =>
-            new TcpEndpoint(Communicator, Protocol, host, port, SourceAddress, timeout, connectionId, compressionFlag);
+            new TcpEndpoint(Communicator,
+                            Protocol,
+                            Transport,
+                            host,
+                            port,
+                            SourceAddress,
+                            timeout,
+                            connectionId,
+                            compressionFlag);
 
         internal virtual ITransceiver CreateTransceiver(
```

Comment:
If the keep a `string transport` parameter, it should be `string transportName`. It would however be preferable to use Transport like everywhere else.

Reply:
It's not clear why we need this parameter at all. 

I think the  CreateTransceiver implementation could check for the IsSecure property value to decide whether or not to create an SslTransceiver or TcpTransceiver.

Reply:
The TransportName is used by WSTransceiver tracing and also in UdpTrasnceiver (not sure why Udp needs it), I guess WSTransceiver can check if the delegate is SslTransceiver and use WSS/WS according to that

Reply:
I remove the param